GeeksCoreLibrary/Modules/PostalServices/PostNL/Services/PostNLService.cs

```diff
@@ -0,0 +1,194 @@
+                    result.CustomerNumber = await objectService.FindSystemObjectByDomainNameAsync("PostNlEuropeCustomerNumber");
+                    result.BarcodeType = "CL";
+                    result.BarcodeSerie = "0000-9999";
+                    result.ProductCode = "4945";
+                    break;
+
+                default:
+                    throw new ArgumentOutOfRangeException(nameof(shippingLocation), shippingLocation, null);
+            }
+
+            return result;
+        }
+
+        /// <summary>
+        /// Creates a new barcode using the PostNL Api
+        /// </summary>
+        /// <param name="orderId">The orderId</param>
+        /// <param name="shippingLocation">The location the order will be send to</param>
+        /// <returns>A model containing the barcode created using the api</returns>
+        public async Task<BarcodeResponseModel> CreateNewBarcode(string orderId, ShippingLocations shippingLocation = ShippingLocations.Netherlands)
+        {
+            var exceptionMessage = "";
+            var responseString = "";
+            var requestString = "";
+
+            try
+            {
+                var restClient = new RestClient(gclSettings.PostNlApiBaseUrl);
+                var restRequest = new RestRequest("/shipment/v1_1/barcode", Method.GET);
+                var settings = await GetSettings(shippingLocation);
+
+                restRequest.AddQueryParameter("CustomerCode", settings.CustomerCode);
+                restRequest.AddQueryParameter("CustomerNumber", settings.CustomerNumber);
+                restRequest.AddQueryParameter("Type", settings.BarcodeType);
+                restRequest.AddQueryParameter("Serie", settings.BarcodeSerie);
+                restRequest.AddHeader("apiKey", gclSettings.PostNlShippingApiKey);
+
+                var response = await restClient.ExecuteAsync<BarcodeResponseModel>(restRequest);
+                responseString = response.Content;
+                if (response.ErrorException != null)
+                {
+                    exceptionMessage = response.ErrorException.ToString();
+                }
+
+                return response.Data;
+            }
+            catch (Exception exception)
+            {
+                exceptionMessage = exception.ToString();
+                throw;
+            }
+            finally
+            {
+                await CleanLogs();
+
+                databaseConnection.ClearParameters();
+                databaseConnection.AddParameter("order_id", orderId);
+                databaseConnection.AddParameter("request", requestString);
+                databaseConnection.AddParameter("response", responseString);
+                databaseConnection.AddParameter("exception", exceptionMessage);
+
+                await databaseConnection.InsertOrUpdateRecordBasedOnParametersAsync("cust_postnl_log", 0UL);
```

Comment:
In de GCL willen we geen tabelnamen direct in query's hebben staan. Kun je hier een constante van maken en dan die constante in de query zetten?